platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -1030,3 +1030,168 @@ def get_sfp(self, index):
             return None
 
         return module.get_sfp(sfp_index - 1)
+
+
+class SmartSwitchChassis(Chassis):
+    def __init__(self):
+        super(SmartSwitchChassis, self).__init__()
+        self.module_initialized_count = 0
+        self.initialize_modules()
+
+    def is_modular_chassis(self):
+        """
+        Retrieves whether the sonic instance is part of modular chassis
+
+        Returns:
+            A bool value, should return False by default or for fixed-platforms.
+            Should return True for supervisor-cards, line-cards etc running as part
+            of modular-chassis.
+            For SmartSwitch platforms this should return True even if they are
+            fixed-platforms, as they are treated like a modular chassis as the
+            DPU cards are treated like line-cards of a modular-chassis.
+        """
+        return True
+
+    ##############################################
+    # Module methods
+    ##############################################
+    def initialize_single_module(self, index):
+        count = self.get_num_modules()
+        if index < count:
+            if not self._module_list:
+                self._module_list = [None] * count
+
+            if not self._module_list[index]:
+                from .module import DpuModule
+                self._module_list[index] = DpuModule(index + 1)
+                self.module_initialized_count += 1
+
+    def initialize_modules(self):
+        if not self._module_list:
```

Comment:
Why can we call here the following:
```
for index in range(count):
    self.initialize_single_module(index)
```
This code duplication is not necessary

Reply:
Fixed